### ISSUE TITLE
rstudio server fails to start session with 

### ISSUE THREAD

**Issue by andreasleha** [rstudio/rstudio#12946]
### System details

    RStudio Edition : Server
    RStudio Version : 2023.03.0+386 (Cherry Blossom) for CentOS 7
    OS Version      : CentOS Linux release 7.9.2009 (Core)
    R Version       : 4.2.3

### Steps to reproduce the problem

1. Start rstudio-server (i.e. via `rstudio-server start` or with more control via `/usr/lib/rstudio-server/bin/rserver --server-daemonize 0`). 
2. Browse to the running rstudio server ip-adrres:8787
3. Log in
4. The error in the web interface is `Could not connect to the R session on RStudio Server. Error occurred during transmision (6)`
5. Logged in `/var/log/messages` is

```
ERROR r error 4 (Error in if (!cranMirrorConfigured) .rs.setCRANRepos(cran, secondary) : #012  missing value where TRUE/FALSE needed#012) [errormsg: Error in if (!cranMirrorConfigured) .rs.setCRANRepos(cran, secondary) : #012  missing value where TRUE/FALSE needed#012]; OCCURRED AT rstudio::core::Error rstudio::r::exec::{anonymous}::evaluateExpressionsUnsafe(SEXP, SEXP, SEXPREC**, rstudio::r::sexp::Protect*, rstudio::r::exec::{anonymous}::EvalType) src/cpp/r/RExec.cpp:198; LOGGED FROM: int rstudio::r::session::RReadConsole(const char*, CONSOLE_BUFFER_CHAR*, int, int) src/cpp/r/session/RStdCallbacks.cpp:299
ERROR R SUICIDE: 2023-03-31T15:04:30.869903Z [rsession-XXXX] ERROR r error 4 (Error in if (!cranMirrorConfigured) .rs.setCRANRepos(cran, secondary) : |||  missing value where TRUE/FALSE needed|||) [errormsg: Error in if (!cranMirrorConfigured) .rs.setCRANRepos(cran, secondary) : |||  missing value where TRUE/FALSE needed|||]; OCCURRED AT rstudio::core::Error rstudio::r::exec::{anonymous}::evaluateExpressionsUnsafe(SEXP, SEXP, SEXPREC**, rstudio::r::sexp::Protect*, rstudio::r::exec::{anonymous}::EvalType) src/cpp/r/RExec.cpp:198; LOGGED FROM: void {anonymous}::rSuicide(const string&) src/cpp/session/SessionMain.cpp:1242
```

### Describe the problem in detail

I do not know which details I could add, but I am happy to produce more detailed information.

### Describe the behavior you expected

The r-session for the user is started and the rstudio interface is shown.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [x] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by andreasleha** [comment]
Thanks for looking into this!  I have an .Rprofile that sets some repositories by default (which is working fine outside of RStudio).  When I move that .Rprofile aside, starting the session with RStudio works.

Seems to be some faulty setting in my .Rprofile and not an issue with RStudio.

**Comment by gtritchie** [comment]
> Seems to be some faulty setting in my .Rprofile and not an issue with RStudio.

Thanks, I will close this.

**Comment by andreasleha** [comment]
Sorry, but I need to re-open this issue.  I still see this problem.  It does not happen every time but every now and again this happens.  And it happens also for users who do not set anything related to repositories in their ~/.Rprofile.

The workaround is to log into the machine and delete the `sessions` folder from `~/.local/share/rstudio`.

Once that is done and the user can login to RStudio again, this is the output of `getOption("repos")`:

```
CRAN

"https://cran.rstudio.com"

BioCsoft

"https://bioconductor.org/packages/3.10/bioc"

BioCann

"https://bioconductor.org/packages/3.10/data/annotation"

BioCexp

"https://bioconductor.org/packages/3.10/data/experiment"

BioCextra

NA
```

**Comment by kevinushey** [comment]
Thanks; this helps -- it looks like something is setting an `NA` repository entry and RStudio isn't handling that.

**Comment by kevinushey** [comment]
Of course, this doesn't answer the question of why some users are seeing an `NA` value in their repos option -- presumedly that's coming from Bioconductor (or a related package) but it's hard to know for sure. RStudio itself doesn't attempt to set the R repositories in this way.

**Comment by andreasleha** [comment]
Great.  Thanks for looking into this.  I am not sure, where this entry comes from.  I can't find it in any of the obvious site-wide configuration files.  We are testing as a workaround to set the repos in one of the Rprofile files.

**Comment by andreasleha** [comment]
Further follow-up on this: RStudio seems to set the `BioCextra NA` repository.  When I start a new R session through RStudio, `getOption("repos")` shows this entry. When I start the same R version on the command line, that entry is not there.  Otherwise the `sessionInfo()` is identical in both cases. 

So, the question now is: why is the "repos" entry different in RStudio and esp. why does it include that repository which causes trouble at the next session start?  And how do I modify that behaviour?

**Comment by kevinushey** [comment]
> So, the question now is: why is the "repos" entry different in RStudio and esp. why does it include that repository which causes trouble at the next session start? And how do I modify that behaviour?

Sorry, but I don't know -- but since (thus far) you're the only person who's reported and been affected by this issue, I suspect it's specific to how R / RStudio is configured on your machine.

**Comment by jonvanausdeln** [comment]
Verified on server 2023.05.0-daily+366
Ubuntu 22, windows11/edge client

Adding the `NA` entry to .Rprofile causes error on Cherry Blossom, but does not on error on this latest dailies build.  

_Note, I'm closing as the originally reported issue is resolved. If need, please open a separate issue for a repository getting set to_ `NA`
